ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java

```diff
@@ -553,11 +553,13 @@ private void release(boolean releaseLocks) {
         LOG.warn("Exception when releasing locking in destroy: " + e.getMessage());
       }
     }
-    ShutdownHookManager.removeShutdownHook(shutdownRunner);
+    ShutdownHookManager.removeShutdownHook(txnRollbackRunner);
   }
 
   void releaseLocksAndCommitOrRollback(boolean commit) throws LockException {
```

Comment:
This method starts to look like a cleanup.
Shall we separate out the functions to a different method, or just change the name?

Reply:
This is the one method that is called from everywhere to end a transaction in either way. Definitely more than a cleanup method. So I think we should not rename it to something that signals it is just a cleanup. I did not want to extract the two lines of clean up to another method either. What is your suggestion?

Reply:
I do not have better idea than separate out to a cleanup method. I was hoping you come up with a golden one :)
Doing cleanup in something called as "releaseLocksAndCommitOrRollback" is definitely not a good idea IMHO.

Reply:
After some more thought, I still was not able to come up with a really good idea :(
The best I can think is that we should rename the method to _commitAndCleanUp_
But I still open for any other ideas if you have some.

Thanks,
Peter

Reply:
I would not rename to `commitAndCleanup` because this method also rolls back transaction. I'll rename it to `endTransactionAndCleanup`